Library/Sender.ts

```diff
@@ -135,7 +135,7 @@ class Sender {
                     if (this._enableDiskRetryMode) {
                         // try to send any cached events if the user is back online
                         if (res.statusCode === 200) {
-                            setTimeout(() => this._sendFirstFileOnDisk(), this._resendInterval);
+                            setTimeout(() => this._sendFirstFileOnDisk(), this._resendInterval).unref();
```

Comment:
would this not cause the process to exist and not retry subsequently?

Reply:
I think this is good behavior of the SDK. If the only reason for it keeping the app running is to check for stored telemetry on the disk every 60 seconds, it should let the app close. Quick closing Node scripts are especially important where you pay for execution time (Lambda / Functions / etc)

Reply:
This only prevents the process from not exiting if it tries to, as per [here](https://nodejs.org/api/timers.html#timers_timeout_unref). After this change, the only thing we block termination on is for pending telemetry in the buffer

Reply:
Thanks for the explanation @OsvaldoRosado , @markwolff . So, this is intentional not to check for pending telemetry if thats the only thing keeping it active.